tests/test_migrations.py

```diff
@@ -303,3 +305,71 @@ def test_migrations_users_sites_foreign_keys(self):
                             for e in meta.tables['users_sites'].foreign_keys]
             for foreign_key in ['users.id', 'sites.id']
         ))
+
+    def test_migrations_patient_consent_date(self):
+        """Test that patient_consent_date is migrated correctly."""
+        engine = self._get_temp_db(10)
+        t_db = TransactionDB(engine)
+        # TODO is this a problem?
+        # TZDateTime apparently does not re-serialize microsceconds correctly
+        now = utcnow().replace(microsecond=0)
+
+        t = Transaction()
+        t.patient_consent = 0
+        t_id_no_consent = t_db.create_transaction(t)
+
+        t = Transaction()
+        t.patient_consent = 1
+        t.data_uploaded = now
+        t_id_date_uploaded = t_db.create_transaction(t)
+
+        # TODO bcause the setter is overwritten in the new version, the legacy
+        # field cannot be set and tested
```

Comment:
Any idea on how to test this? I would need a t.db version with a `patient_consent` field, set that without settings `data_uploaded` and then migrate it. But since the SQLAlchemy model is already on the new version, it's using the proxy setter when trying to set the legacy field

Reply:
See comment above